osu/map.py

```diff
@@ -52,9 +56,30 @@ def files(self):
 
     def zip(self):
         map_pack_file = MAP_PACKS_FOLDER.joinpath(f"{uuid.uuid4()}.zip")
-
+        print(map_pack_file)
         files = self.files()
         FileTools.zip_files(files, to_file=map_pack_file)
-        FileTools.delete_files(files)
 
         return map_pack_file
+
+    async def upload(self) -> str | None:
+
+        await DbxClient.init(DROPBOX_ACCESS_TOKEN)
+
+        if DbxClient.client is None:
+            raise ValueError("DbxClient is not connected")
+
+        zf = self.zip()
+        dbx_destination = DbxClient.dbx_path + zf.name
+        try:
+            with open(zf, "rb") as f:
+                DbxClient.client.files_upload(f.read(), dbx_destination)
+                shared_link_metadata = DbxClient.client.sharing_create_shared_link_with_settings(dbx_destination)
+                #  print(f"File uploaded successfully! {shared_link_metadata.url} is url")
+                # Temporary file deletion
+                files = self.files()
+                FileTools.delete_files(files)
+                return shared_link_metadata.url
+        except Exception as e:
+            print(f"Error uploading: {e}")
+            return None # thoughts on throwing an error here instead?
```

Comment:
i think either is okay.

strictly speaking i think it should be an error becausw something essentail failed, then we can return a message to the user if an exception is caught

Reply:
catching exceptions now, will make it more specific later on